### PR TITLE
#1498 - fix the processing of automation resource queue

### PR DESCRIPTION
**Pull request by adimova**
Signed-off-by: Ana Dimova <a.dimova@prosyst.bg>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java [hunk 1]
```diff
@@ -302,8 +302,8 @@ protected void processAutomationProvider(Bundle bundle) {
         try {
             urlEnum = bundle.findEntries(path, null, false);
         } catch (IllegalStateException e) {
-            logger.debug("Can't read from resource of bundle with ID " + bundle.getBundleId()
-                    + ". The bundle is uninstalled.", e);
+            logger.debug(String.format("Can't read from resource of bundle with ID {}. The bundle is uninstalled.",
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java]: why do you use String.format here? Formatting is directly done by the parameterized logging.

File: bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java [hunk 2]
```diff
@@ -324,7 +324,8 @@ protected void processAutomationProvider(Bundle bundle) {
                     try {
                         importData(vendor, parser, reader = new InputStreamReader(url.openStream()));
                     } catch (IOException e) {
-                        logger.error("Can't read from resource of bundle with ID " + bundle.getBundleId(), e);
+                        logger.error(
+                                String.format("Can't read from resource of bundle with ID {}", bundle.getBundleId()), e);
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java]: same here

File: bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AutomationResourceBundlesEventQueue.java [hunk 3]
```diff
@@ -160,7 +160,14 @@ public void run() {
             }
             Iterator<BundleEvent> events = l_queue.iterator();
             while (events.hasNext()) {
-                processBundleChanged(events.next());
+                BundleEvent event = events.next();
+                try {
+                    processBundleChanged(event);
+                } catch (Throwable t) {
+                    String msg = String.format("Processing bundle event {}, for automation resource bundle '{}' failed",
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AutomationResourceBundlesEventQueue.java]: same here

**Comment by kaikreuzer** [comment]
Could you please change all occurrences of String.format to [parametrized logging](http://www.slf4j.org/faq.html#paramException)?

**Comment by adimova** [comment]
updated

**Comment by kaikreuzer** [comment]
Thanks!

**Pull request merged**